example/global.cc

```diff
@@ -31,164 +31,134 @@ void check(T actual, U expected) {
-    exit(1);
-  }
+  auto file = load_file("global.wasm");
 
   // Create external globals.
   std::cout << "Creating globals..." << std::endl;
-  auto const_f32_type = wasm::GlobalType::make(
-    wasm::ValType::make(wasm::ValKind::F32), wasm::Mutability::CONST);
-  auto const_i64_type = wasm::GlobalType::make(
-    wasm::ValType::make(wasm::ValKind::I64), wasm::Mutability::CONST);
-  auto var_f32_type = wasm::GlobalType::make(
-    wasm::ValType::make(wasm::ValKind::F32), wasm::Mutability::VAR);
-  auto var_i64_type = wasm::GlobalType::make(
-    wasm::ValType::make(wasm::ValKind::I64), wasm::Mutability::VAR);
-  auto const_f32_import = wasm::Global::make(store, const_f32_type.get(), wasm::Val::f32(1));
-  auto const_i64_import = wasm::Global::make(store, const_i64_type.get(), wasm::Val::i64(2));
-  auto var_f32_import = wasm::Global::make(store, var_f32_type.get(), wasm::Val::f32(3));
-  auto var_i64_import = wasm::Global::make(store, var_i64_type.get(), wasm::Val::i64(4));
+  auto var_f32_import = store.make_global(3.0f);
+  auto var_i64_import = store.make_global(uint64_t(4));
 
   // Instantiate.
   std::cout << "Instantiating module..." << std::endl;
-  auto imports = wasm::vec<wasm::Extern*>::make(
-    const_f32_import.get(), const_i64_import.get(),
-    var_f32_import.get(), var_i64_import.get()
-  );
-  auto instance = wasm::Instance::make(store, module.get(), imports);
-  if (!instance) {
-    std::cout << "> Error instantiating module!" << std::endl;
+  wasm::Imports imports{
+    {
+      1.0f,
+      2.0,
+      var_f32_import,
+      var_i64_import,
+    },
+  };
+  auto instance = store.instantiate(file, imports);
+  if (instance.error()) {
+    std::cout << "> Error instantiating module: " << error->what() << std::endl;
     exit(1);
   }
 
   // Extract export.
   std::cout << "Extracting exports..." << std::endl;
   auto exports = instance->exports();
   size_t i = 0;
-  auto const_f32_export = get_export_global(exports, i++);
-  auto const_i64_export = get_export_global(exports, i++);
-  auto var_f32_export = get_export_global(exports, i++);
-  auto var_i64_export = get_export_global(exports, i++);
-  auto get_const_f32_import = get_export_func(exports, i++);
-  auto get_const_i64_import = get_export_func(exports, i++);
-  auto get_var_f32_import = get_export_func(exports, i++);
-  auto get_var_i64_import = get_export_func(exports, i++);
-  auto get_const_f32_export = get_export_func(exports, i++);
-  auto get_const_i64_export = get_export_func(exports, i++);
-  auto get_var_f32_export = get_export_func(exports, i++);
-  auto get_var_i64_export = get_export_func(exports, i++);
-  auto set_var_f32_import = get_export_func(exports, i++);
-  auto set_var_i64_import = get_export_func(exports, i++);
-  auto set_var_f32_export = get_export_func(exports, i++);
-  auto set_var_i64_export = get_export_func(exports, i++);
+  auto const_f32_export = instance->exported_global(i++);
```

Comment:
Is it ever possible to have an `exports` without an `Instance`?

Reply:
No, should it be? I don't think so, but if we should then what would create one?